paimon-core/src/main/java/org/apache/paimon/catalog/CachingCatalog.java

```diff
@@ -309,32 +298,29 @@ private class TableInvalidatingRemovalListener implements RemovalListener<Identi
         public void onRemoval(Identifier identifier, Table table, @NonNull RemovalCause cause) {
             LOG.debug("Evicted {} from the table cache ({})", identifier, cause);
             if (RemovalCause.EXPIRED.equals(cause)) {
-                tryInvalidateSysTables(identifier);
+                tryInvalidateAttachedTables(identifier);
             }
         }
     }
 
     @Override
     public void invalidateTable(Identifier identifier) {
         tableCache.invalidate(identifier);
-        tryInvalidateSysTables(identifier);
+        tryInvalidateAttachedTables(identifier);
         if (partitionCache != null) {
             partitionCache.invalidate(identifier);
         }
     }
 
-    private void tryInvalidateSysTables(Identifier identifier) {
-        if (!isSpecifiedSystemTable(identifier)) {
-            tableCache.invalidateAll(allSystemTables(identifier));
-        }
-    }
-
-    private static Iterable<Identifier> allSystemTables(Identifier ident) {
-        List<Identifier> tables = new ArrayList<>();
-        for (String type : SYSTEM_TABLES) {
-            tables.add(Identifier.fromString(ident.getFullName() + SYSTEM_TABLE_SPLITTER + type));
+    private void tryInvalidateAttachedTables(Identifier identifier) {
+        if (identifier.isMainTable()) {
```

Comment:
We don't need to care about branch tables.

Reply:
Why, when the table is deleted, the cached branch table should also be cleaned up.

Reply:
Maybe we can just do this in `invalidateTable`?

Reply:
I see, we can move this logic to drop table

Reply:
Alter table with branch is also supported, maybe still in `invalidateTable`?

Reply:
But not in `TableInvalidatingRemovalListener`.